gateway-applications/src/main/resources/applications/tokengen/app/js/tokengen.js

```diff
@@ -53,20 +53,23 @@ function b64DecodeUnicode(str) {
 }
 
 var gen = function() {
+	$('#invalidLifetimeText').hide();
     var pathname = window.location.pathname;
     var topologyContext = pathname.replace(loginPageSuffix, "");;
     var baseURL = topologyContext.substring(0, topologyContext.lastIndexOf('/'));
     baseURL = baseURL.substring(0, baseURL.lastIndexOf('/') + 1);
     var tokenURL = topologyContext + knoxtokenURL;
     var form = document.forms[0];
     //var comment = form.comment.value;
-    var lifespan = form.lifespan.value;
+    var lt_days = form.lt_days.value;
+    var lt_hours = form.lt_hours.value;
+    var lt_mins = form.lt_mins.value;
     var _gen = function() {
         var apiUrl = tokenURL;
         //Instantiate HTTP Request
-        var params = '';
-        if (lifespan != '') {
-        	params = '?lifespan=' + lifespan;
+        var params = '?lifespan=P' + lt_days + "DT" + lt_hours + "H" + lt_mins + "M";  //we need to support Java's Duration pattern
```

Comment:
So, this would end up with a query param like: "1DT2H3M"?
Why are we using DT instead of D?
We are also going to be exposing this as a general API.
Wondering whether we would like to be able to just say 8H for 8 hours instead of "0DT8H0M".

Reply:
Thanks, @lmccay for your review.
The required pattern by Java's [Duration.parse](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-) is :`PnDTnHnMn.nS`.
Since I did not want to reinvent the wheel, we could re-use Java's own public API. Any objection?

Reply:
@lmccay - I'm going to merge this change as-is, if you want we can open a follow-up JIRA with a different `queryParam` language later.